src/Riok.Mapperly/Helpers/SymbolExtensions.cs

```diff
@@ -79,6 +79,49 @@ internal static IEnumerable<IPropertySymbol> GetAllAccessibleProperties(this ITy
         return genericIntf != null;
     }
 
+    internal static bool ImplementsGeneric(
+        this ITypeSymbol t,
+        INamedTypeSymbol genericInterfaceSymbol,
+        string symbolName,
+        [NotNullWhen(true)] out INamedTypeSymbol? genericIntf,
+        out bool isExplicit)
+    {
+        if (SymbolEqualityComparer.Default.Equals(t.OriginalDefinition, genericInterfaceSymbol))
+        {
+            genericIntf = (INamedTypeSymbol)t;
+            isExplicit = true;
+            return true;
+        }
+
+        genericIntf = t.AllInterfaces.FirstOrDefault(x => x.IsGenericType && SymbolEqualityComparer.Default.Equals(x.OriginalDefinition, genericInterfaceSymbol));
+
+        if (genericIntf == null)
+        {
+            isExplicit = false;
+            return false;
+        }
+
+        var interfaceSymbol = genericIntf.GetMembers(symbolName).Single();
+
+        var symbolImplementaton = t.FindImplementationForInterfaceMember(interfaceSymbol);
+
+        // if null then the method is unimplemented
+        // symbol implements genericInterface but has not implemented the corresponding methods
+        // this can only occur in unit tests
+        if (symbolImplementaton == null)
+            throw new NotSupportedException("Symbol implementation cannot be null for objects implementing interface.");
+
+        // check if symbol is explicit
+        isExplicit = symbolImplementaton switch
+        {
+            IMethodSymbol methodSymbol => !methodSymbol.ExplicitInterfaceImplementations.Any(),
+            IPropertySymbol propertySymbol => !propertySymbol.ExplicitInterfaceImplementations.Any(),
+            _ => throw new NotImplementedException(),
+        };
+
+        return true;
+    }
+
     internal static bool HasImplicitInterfaceMethod(this ITypeSymbol symbol, INamedTypeSymbol inter, string methodName)
```

Comment:
This could now be simplified a lot, couldn't it? (eg. `symbol.ImplementsGeneric(inter, methodName, out var isExplicit) && !isExplicit`